src/main/java/org/folio/fqm/repository/DataRefreshRepository.java

```diff
@@ -0,0 +1,162 @@
+
+  private final SimpleHttpClient simpleHttpClient;
+
+  public void refreshMaterializedViews(String tenantId) {
+    for (String matViewName : MATERIALIZED_VIEW_NAMES) {
+      String fullName = tenantId + "_mod_fqm_manager." + matViewName;
+      log.info("Refreshing materialized view {}", fullName);
+      jooqContext.execute(REFRESH_MATERIALIZED_VIEW_SQL + fullName);
+    }
+  }
+
+  public void refreshExchangeRates(String tenantId) {
+    log.info("Refreshing exchange rates");
+    String fullTableName = tenantId + "_mod_fqm_manager.currency_exchange_rates";
+    String systemCurrency = getSystemCurrencyCode();
+    if (!SYSTEM_SUPPORTED_CURRENCIES.contains(systemCurrency)) {
+      log.info("System currency does not support automatic exchange rate calculation");
+      return;
+    }
+
+    List<Record2<String, Double>> exchangeRates = new ArrayList<>();
+    for (String currency : SYSTEM_SUPPORTED_CURRENCIES) {
+      Double exchangeRate = getExchangeRate(currency, systemCurrency);
+      if (!Double.isNaN(exchangeRate)) {
+        Record2<String, Double> currencyExchangeRate = jooqContext
+          .newRecord(CURRENCY_FIELD, EXCHANGE_RATE_FIELD)
+          .value1(currency)
+          .value2(exchangeRate);
+        exchangeRates.add(currencyExchangeRate);
+      }
+    }
+
+    jooqContext.insertInto(table(fullTableName), CURRENCY_FIELD, EXCHANGE_RATE_FIELD)
+      .valuesOfRecords(exchangeRates)
+      .onConflict(CURRENCY_FIELD)
+      .doUpdate()
+      .set(EXCHANGE_RATE_FIELD, DSL.field("EXCLUDED." + EXCHANGE_RATE_FIELD.getName(), Double.class))
+      .execute();
+  }
+
+  private String getSystemCurrencyCode() {
+    log.info("Getting system currency");
+    try {
+      String localeSettingsResponse = simpleHttpClient.get(GET_LOCALE_SETTINGS_PATH, GET_LOCALE_SETTINGS_PARAMS);
+      ObjectMapper objectMapper = new ObjectMapper();
+      JsonNode localeSettingsNode = objectMapper.readTree(localeSettingsResponse);
+      String valueString = localeSettingsNode
+        .path("configs")
+        .get(0)
+        .path("value")
+        .asText();
+      JsonNode valueNode = objectMapper.readTree(valueString);
+      return valueNode.path("currency").asText();
+    } catch (Exception e) {
+      log.info("No system currency defined, defaulting to USD");
+      return "USD";
+    }
+  }
+
+  private Double getExchangeRate(String fromCurrency, String toCurrency) {
+    log.info("Getting currency exchange rate from {} to {}", fromCurrency, toCurrency);
+    Map<String, String> exchangeRateParams = Map.of(
+      "from", fromCurrency,
+      "to", toCurrency
+    );
+    try {
+      String exchangeRateResponse = simpleHttpClient.get(GET_EXCHANGE_RATE_PATH, exchangeRateParams);
+      DocumentContext exchangeRateInfo = JsonPath.parse(exchangeRateResponse);
+      var exchangeRate = exchangeRateInfo.read("exchangeRate");
+      if (exchangeRate instanceof BigDecimal bd) {
+        return bd.doubleValue();
+      }
```

Comment:
the /exchange-rates API can return a double or a bigdecimal, so we have to check for that here.